### PR TITLE
HDDS-5289. Update container's deleteTransactionId on creation of the transaction in SCM.

### PR DESCRIPTION
**Pull request by lokeshj1703**
## What changes were proposed in this pull request?

Container's delete transactionId is updated today when block deleting service processes the transaction. This makes sure that container's transactionId is in sync with replicas. 
With HA updating transactionId could lead to several ratis log entries. The PR makes a change where txnId for container is updated when transaction is created in SCM rather than when it is processed.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-5289

## How was this patch tested?

Existing UT

### CODE REVIEW

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogStateManagerImpl.java [hunk 1]
```diff
@@ -167,8 +172,13 @@ public void removeFromDB() throws IOException {
   @Override
   public void addTransactionsToDB(ArrayList<DeletedBlocksTransaction> txs)
       throws IOException {
+    Map<ContainerID, Long> containerIdToTxnIdMap = new HashMap<>();
     for (DeletedBlocksTransaction tx : txs) {
+      long tid = tx.getTxID();
+      containerIdToTxnIdMap.compute(ContainerID.valueOf(tx.getContainerID()),
+          (k, v) -> v != null && v > tid ? v : tid);
       transactionBuffer.addToBuffer(deletedTable, tx.getTxID(), tx);
```
Comment (bshashikant) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogStateManagerImpl.java:180]: transactionBuffer.addToBuffer will update the DB only when the ratis snapshot is taken, where as the next call  , containerManager.updateDeleteTransactionId will update the in-memory state as well the container db right away. Is this desirable?

**Review by bshashikant — COMMENTED** [review]
Can you please add unit tests for this in SCM HA environment?

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogStateManagerImpl.java (hunk 1 above)
Comment (lokeshj1703) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogStateManagerImpl.java:180]: Fixed in latest commit. containerManager.updateDeleteTransactionId  now updates DB via transactionBuffer.

**Review by bshashikant — APPROVED** [review]

**Comment by bshashikant** [comment]
Thanks @lokeshj1703 for the fix.

**Pull request merged**